src/index.ts

```diff
@@ -1,48 +1,98 @@
+     checkHandlerName() {
+          this.serverless.cli.log('Checking handler names');
 
           const functionNames = this.serverless.service.getAllFunctions();
           functionNames.forEach(fnName => {
-               // Get handler name
                const fn = this.serverless.service.getFunction(fnName) as Serverless.FunctionDefinitionHandler;
-               const handler = fn.handler.split('/').pop()?.slice(0, -8) as string;
+               // Get the full handler name (including path)
+               const handlerFullPath = fn.handler as string;
+               // Get the handler name with file extension but no path
+               const handlerName = handlerFullPath.split('/').pop() as string;
+               // Get the handler name with no path and no file extension
+               const handlerNameNoExtension = handlerName.replace('.handler', '') as string;
 
-               // Convert the function name and handler into values that can be compared (string[])
-               // Convert from camel case to an array of strings
-               const splitFn = fnName.split(/(?=[A-Z])/).map(s => s.toLowerCase());
-               // Convert from - to an array of strings
-               const splitHandler = handler.split('-');
+               // Check that the handler name is in kebab case
+               if (handlerNameNoExtension !== kebabCase(handlerNameNoExtension)) {
+                    this.errors.push(`Warning: Handler "${handlerName}" is not kebab case (dash delimited)`);
+               }
 
-               // Throw an error if the function does not match the handler name
-               if (JSON.stringify(splitFn) != JSON.stringify(splitHandler)) {
-                    // this.serverless.cli.log(`Warning: Function "${fnName}" does not match handler name "${handler}.handler"`);
-                    throw new Error(`Warning: Function "${fnName}" does not match handler name "${handler}.handler"`);
+               // Check that the handler ends in .handler
+               if (handlerNameNoExtension === handlerName) {
+                    this.errors.push(`Warning: Handler "${handlerName}" does not end in ".handler"`);
                }
           });
+     }
 
-          this.serverless.cli.log('Function check complete! No errors were found.');
+     // Function name conventions
+     // Must be camelCase
+     checkFunctionName() {
+          this.serverless.cli.log('Checking function names');
+
+          const functionNames = this.serverless.service.getAllFunctions();
+          functionNames.forEach(fnName => {
+               // Check that the function name is in camel case
+               if (fnName !== camelCase(fnName)) {
+                    this.errors.push(`Warning: Function "${fnName}" is not camel case`);
+               }
+          });
+     }
+
+     // Function / handler name conventions
+     // Handler names must have the same name as the function
+     checkHandlerNameMatchesFunction() {
```

Comment:
I think it would be worth adding a couple test cases for these functions. Happy for this to happen latter. 

However, I suggest a small piece of refactoring now in preparation.

You will find that this structure will be difficult to test. What I would suggest is make the functions more "functional".
By this I mean, they should take an array of `FunctionDefinitionHandler` and returns an array of string (the error messages).

i.e:

`checkHandlerNameMatchesFunction(handlers: FunctionDefinitionHandler) : string`